### PR TITLE
Add how to add filters to tab menu example see issue #2607

### LINKED ISSUE #2607: Docs improvement how to add filters to tab menu
Since the tab menu is part of the list action it should be mentioned in the docs that you can use the `configureTabMenu` method for adding filters. For instance:

```php
use Sonata\AdminBundle\Admin\Admin;
use Knp\Menu\ItemInterface as MenuItemInterface;
use Sonata\CoreBundle\Form\Type\EqualType;

class FooAdmin extends Admin
{
    protected function configureTabMenu(MenuItemInterface $menu, $action, AdminInterface $childAdmin = null)
    {
        if (!$childAdmin && !in_array($action, array('edit', 'show', 'list'))) {
            return;
        }

        if ($action == 'list') {
            $filterParameters = $this->getFilterParameters();

            // Add or override filter parameters
            $filterParameters['bar'] = array(
                        'type'  => EqualType::TYPE_IS_EQUAL, // equal to
                        'value' => 'baz'
                    );

            // Add filters to uri of tab
            $menu->addChild('Show Baz only', array('uri' => $this->generateUrl('list', array(
                    'filter' => $filterParameters
                ))));

            return;
        }
    }
}
```

Should I create a PR for this section?
http://sonata-project.org/bundles/admin/master/doc/reference/advanced.html#dropdowns-in-tab-menu

It could also be added to the custom `CRUDController` chapters:
- http://sonata-project.org/bundles/doctrine-orm-admin/master/doc/tutorial/creating_your_first_admin_class/defining_crud_controller.html. 
- http://sonata-project.org/bundles/admin/master/doc/cookbook/recipe_custom_action.html

Example usage:
```php
return new RedirectResponse($this->admin->generateUrl('list', array('filter' => $this->admin->getFilterParameters())));
```
Source: http://russische-musik.vs137150.vserver.de/v/Sonata/AdminBundle/Controller/CRUDController:addFlash

Related to:
- https://github.com/sonata-project/SonataAdminBundle/issues/2529
- https://github.com/sonata-project/SonataAdminBundle/issues/1556
- https://github.com/sonata-project/SonataAdminBundle/issues/1556

### PR DESCRIPTION
**Pull request by webdevilopers**

### CODE REVIEW

File: Resources/doc/reference/advanced.rst [hunk 1]
```diff
@@ -230,6 +230,51 @@ You will just need to change the way forms are configured in order to take into
...
+            }
+
+            if ($action == 'list') {
+                // Get current filter parameters
+                $filterParameters = $this->getFilterParameters();
+
+                // Add or override filter parameters
+                $filterParameters['status'] = array(
+                        'type'  => EqualType::TYPE_IS_EQUAL, // => 1
+                        'value' => Delivery::STATUS_OPEN
```
Comment (Soullivaneuh) [inline Resources/doc/reference/advanced.rst]: Missing comma at end of line.

File: Resources/doc/reference/advanced.rst [hunk 2]
```diff
@@ -230,6 +230,51 @@ You will just need to change the way forms are configured in order to take into
...
+                    );
+
+                // Add filters to uri of tab
+                $menu->addChild('List open deliveries', array('uri' => $this->generateUrl('list', array(
+                        'filter' => $filterParameters
```
Comment (Soullivaneuh) [inline Resources/doc/reference/advanced.rst]: Missing comma at end of line and to much indentation.

File: Resources/doc/reference/advanced.rst [hunk 3]
```diff
@@ -230,6 +230,51 @@ You will just need to change the way forms are configured in order to take into
...
+                return;
+            }
+
+            if ($action == 'list') {
+                // Get current filter parameters
+                $filterParameters = $this->getFilterParameters();
+
+                // Add or override filter parameters
+                $filterParameters['status'] = array(
+                        'type'  => EqualType::TYPE_IS_EQUAL, // => 1
```
Comment (Soullivaneuh) [inline Resources/doc/reference/advanced.rst]: Remove one indentation please.

**Comment by Soullivaneuh** [comment]
Could you please rebase your PR?

**Comment by webdevilopers** [comment]
Havn't done a rebase yet, @Soullivaneuh . Can you tell me how to do it on github?
Guess my fork has to update to the current `dev-master`?

**Comment by Soullivaneuh** [comment]
This trick may help you: http://stackoverflow.com/a/17182696/1731473

**Comment by webdevilopers** [comment]
I must confess that I created the patch on the fly on github. I don't have any local branches on my current machine. Since this is only a single file, could you add the patch resp. create a new PR @Soullivaneuh ?

**Comment by Soullivaneuh** [comment]
You just have to clone you fork on local and checkout to `patch-1` branch.

**Comment by core23** [comment]
Can you please rebase @webdevilopers

**Comment by OskarStark** [comment]
Created a new PR in #3743

**Pull request closed without merging**